neqo-http3/src/features/extended_connect/webtransport_session.rs

```diff
@@ -279,6 +286,28 @@ impl WebTransportSession {
                             status,
                             headers,
                         );
+                        // > WebTransport endpoints SHOULD buffer streams and
+                        // > datagrams until they can be associated with an
+                        // > established session.
+                        //
+                        // <https://www.ietf.org/archive/id/draft-ietf-webtrans-http3-13.html#section-4.5>
+                        #[expect(
+                            clippy::iter_over_hash_type,
+                            reason = "no defined order necessary"
+                        )]
+                        for stream_id in pending_streams.drain() {
+                            self.events.extended_connect_new_stream(
+                                Http3StreamInfo::new(
+                                    stream_id,
+                                    ExtendedConnectType::WebTransport
+                                        .get_stream_type(self.session_id),
+                                ),
+                                // Explicitly emit a stream readable event. Such
+                                // event was previously suppressed as the
+                                // session was still negotiating.
+                                true,
```

Comment:
Are you sure that the stream truly is readable?  It is possible to have a stream created without data being sent.  Out of order data or flow control streams cause the stream to be created, but not be readable.

Reply:
Good point. Did some debugging. Turns out, we always emit a `DataReadable` on a new stream, whether the stream is actually readable or not.

I suggest tackling this in a separate pull request. I don't think it is high priority.

Tracked in https://github.com/mozilla/neqo/issues/2916.